requests/dialer_swarm.go

```diff
@@ -243,3 +260,35 @@ func (e *ErrorDial) Error() string {
 func (e *ErrorDial) Unwrap() error {
 	return e.Cause
 }
+
+type dialerWithDNSCache struct {
+	dialer *net.Dialer
+}
+
+func wrapDialer(dialer *net.Dialer) *dialerWithDNSCache {
+	return &dialerWithDNSCache{
+		dialer: dialer,
+	}
+}
+
+func (wd *dialerWithDNSCache) DialContext(ctx context.Context, network, addr string) (net.Conn, error) {
+	go func() {
+		if !isIP(addr) {
```

Comment:
Nitpicking: We could avoid creating the goroutine if we do this check first. It's cheap though, so no big deal. But I have no context yet on how many of these dials are being done, so might be worth considering.

Reply:
This is very tiny save here. Let's keep it as is until we will have a need for it.